sdk/spring-sdk/src/test/scala/kalix/springsdk/impl/action/ActionIntrospectorSpec.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl.action
+
+import scala.reflect.ClassTag
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.protobuf.Descriptors.FieldDescriptor.JavaType
+import kalix.javasdk.action.Action
+import kalix.springsdk.action.RestAnnotatedActions.DeleteWithOneParam
+import kalix.springsdk.action.RestAnnotatedActions.GetClassLevel
+import kalix.springsdk.action.RestAnnotatedActions.GetWithOneParam
+import kalix.springsdk.action.RestAnnotatedActions.GetWithoutParam
+import kalix.springsdk.action.RestAnnotatedActions.PatchWithOneParam
+import kalix.springsdk.action.RestAnnotatedActions.PatchWithoutParam
+import kalix.springsdk.action.RestAnnotatedActions.PostWithOneParam
+import kalix.springsdk.action.RestAnnotatedActions.PostWithTwoMethods
+import kalix.springsdk.action.RestAnnotatedActions.PostWithTwoParam
+import kalix.springsdk.action.RestAnnotatedActions.PostWithoutParam
+import kalix.springsdk.action.RestAnnotatedActions.PutWithOneParam
+import kalix.springsdk.action.RestAnnotatedActions.PutWithoutParam
+import kalix.springsdk.impl.reflection.NameGenerator
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class ActionIntrospectorSpec extends AnyWordSpec with Matchers {
+
+  "Action introspector" should {
```

Comment:
I wrote this test to try to reproduce the issue I saw yesterday, but without success. Which is good news. For completeness I just expand the test to exercise some other users cases. 

I do think that we will need to improve this and produce more clear error messages. This test will be a good place to verify it, but can be done later on.